RCNN_tf.py

```diff
@@ -116,209 +176,292 @@ def loop_body(x, recurrent_states, output):
+
+    Returns:
+        output       (tf.Tensor): output tensor of the softmax layer
+        linear_trans (tf.Tensor): the linear transformation of the input tensor,
+                                  multiplied by the weights plus the bias
+    """
     weights = tf.Variable(
-        tf.truncated_normal([num_input_dim, num_output_dim], stddev=std),
+        tf.truncated_normal([num_input_dim, num_output_dim], stddev=stddev),
         trainable=True,
         name=name + '_weights'
     )
 
     bias = tf.Variable(
-        tf.truncated_normal([num_output_dim], stddev=std),
+        tf.truncated_normal([num_output_dim], stddev=stddev),
         trainable=True,
         name=name + '_bias'
     )
 
     linear_trans = tf.matmul(input_data, weights) + bias
-    return tf.nn.softmax(linear_trans), linear_trans
+    output = tf.nn.softmax(linear_trans)
+
+    return output, linear_trans
 
 
-def accuracy(
-        labels,
-        result
-):
+def accuracy(labels, result):
+    """Computes the accuracy of the classifier
+
+    Args:
+        labels  (tf.Tensor): the true labels
+        results (tf.Tensor): the predicted labels
+
+    Returns:
+        acc     (tf.Tensor): the accuracy of the classifier
+    """
     correct_prediction = tf.equal(tf.argmax(labels, 1), tf.argmax(result, 1))
-    return tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+    acc = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+
+    return acc
 
 
 class RCNN:
+    """The Recurrent Convolutional Neural Network Classifier
+
+    Attributes:
+        input_shape        (list): specification of the image input shape
+        output_shape       (list): specification of the class output shape
+        processing_unit     (str): specification of the PU to be used
+        learning_rate     (float): learning rate
+        num_filter          (int): number of filters to be used
+        shuf_buf_size       (int): buffer size for shuffling
+        conv_filter_shape  (list): specification of conv filter shape
+        rconv_filter_shape (list): specification of recurrent conv filter shape
+        pool_shape         (list): specification of pooling shape
+        pool_stride_shape  (list): spec of pooling stride shape
+    """
     def __init__(
             self,
             input_shape,
-            device_name="/cpu:0",
             output_shape=[None, 10],
+            processing_unit="/cpu:0",
             learning_rate=.1,
             num_filter=64,
-            shuffle_buffer_size=10000,
-            conv_layer_filter_shape=[5, 5],
-            rec_conv_layer_filter_shape=[3, 3],
-            pooling_shape=[3, 3],
-            pooling_stride_shape=[2, 2]
-    ):
+            shuf_buf_size=10000,
+            conv_filter_shape=[5, 5],
+            rconv_filter_shape=[3, 3],
+            pool_shape=[3, 3],
+            pool_stride_shape=[2, 2]
+        ):
+
+        # Input and output shape placeholders
```

Comment:
That's not true. These are placeholders for the input and output data. However, the known dimensions should be set beforehand. This is why the shape is explicitly set. Since we want to be flexible with the number of inputs we can pass `None`

Reply:
Good!